File: 49/Solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        anagrams = dict()
        for word in strs:
            x = ''.join(sorted([w for w in word]))
            if anagrams.get(x):
                anagrams[x].append(word)
            else:
                anagrams[x] = [word]
                
        return anagrams.values()

File: 49/Solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> hmap = new HashMap<String, List<String>>();
        
        List<List<String>> ls = new ArrayList<List<String>>();
        
        for(String str: strs) {
            
            char []c = str.toCharArray();
            Arrays.sort(c);      
            String x = new String(c);
            
            if(hmap.get(x) != null) {
                List<String> y = hmap.get(x); 
                y.add(str);
                hmap.put(x, y);
            } 
            else {
                List<String> y = new ArrayList<String>(); 
                y.add(str);
                hmap.put(x, y);
            }
            
        }
        
        for(String s: hmap.keySet()) {
            ls.add(hmap.get(s));
        }
        
        
        return ls;
    }
}